Aio readme (#155)
Azure/arc_jumpstart_docs: docs/azure_arc_jumpstart/azure_edge_iot_ops/aio_manufacturing/_index.md

---
type: docs

> title: "Azure Edge IoT Operations - Manufacturing"
> linkTitle: "Azure Edge IoT Operations - Manufacturing"

Reviewer: ```suggestion
title: "Manufacturing with Azure IoT Operations"
linkTitle: "Azure Manufacturing with Azure IoT Operations"
```

Reviewer: It's "Azure IoT Operation" no "Edge". Change across the doc

weight: 3
description: >
---

## Enhance operational insights at the edge using Azure Edge IoT Operations (AIO)

> Contoso Bakeries, a fictional company, requires dashboards and analytics on the Edge (in its plant) and in Azure to provide real-time operational data to plant operators such as Operational Technology (OT) Managers, Operators, and Maintenance personnel. This includes insights into production line operations and key performance indicators (KPIs) like overall equipment effectiveness (OEE), downtime, and waste. A prime focus is on data from critical equipments like fryers. Additionally, regional executives and directors are interested in supervising main KPIs across multiple plants remotely.

Reviewer: ```suggestion
Contoso Bakeries, a fictional company, requires dashboards and analytics on the edge (in its plant) and in Azure to provide real-time operational data to plant operators such as Operational Technology (OT) Managers, Operators, and Maintenance personnel. This includes insights into production line operations and key performance indicators (KPIs) like overall equipment effectiveness (OEE), downtime, and waste. A prime focus is on data from critical equipment like fryers. Additionally, regional executives and directors are interested in supervising main KPIs across multiple plants remotely.
```

## Architecture

> The solution for Contoso will involve deploying the following:
> 
> - A Kubernetes cluster on the Edge using AKS Edge Essentials, managed by Azure Arc, which hosts various Azure IoT Operations services.
> - A custom-developed MQTT-Simulator that mimics plant and fryer data, transmitting it to Azure IoT MQ via MQTT.
> - An MQTT-listener service that captures this production line data, sending it to an open-source InfluxDb on the Edge, generating near real-time dashboards and analytics and visualizes it using Grafana.
> - An Azure IoT MQ connection with Azure Event Grid namespace, an MQTT broker integrated into Azure, pushing data to Azure Event Grid and subsequently to Azure Data Explorer.
> - Azure Data Explorer dashboards and analytics for supervisory and executive roles monitoring multiple plants' operations.

Reviewer: ```suggestion
The solution for Contoso will involve deploying the following:
- A Kubernetes cluster on the edge using AKS Edge Essentials, managed by Azure Arc, which hosts various Azure IoT Operations (AIO) services.
- A custom-developed MQTT-Simulator that mimics plant and fryer data, transmitting it to Azure IoT MQ via MQTT.
- An MQTT-listener service that captures this production line data, sending it to an open-source InfluxDb on the Edge, generating near real-time dashboards and analytics, and visualizing it using Grafana.
- An Azure IoT MQ connection with Azure Event Grid namespace, an MQTT broker integrated into Azure, pushing data to Azure Event Grid and subsequently to Azure Data Explorer.
- Azure Data Explorer dashboards and analytics for supervisory and executive roles monitoring multiple plants' operations.
```

Reviewer: - No need to capitalize Edge in a sentence
- Missing link to AIO product docs, InfluxDB, Azure Data Explorer (check more)
- open source - no hyphen

> ![Architecture diagram](./01.png)

Reviewer: ```suggestion
![Screenshot of the scenario architecture diagram](./01.png)
```

> The following Jumpstart scenario will show how to create an AKS Edge Essentials cluster in Azure Windows Server VM and connect the Azure VM and AKS Edge Essentials cluster to Azure Arc using [Azure Bicep](https://learn.microsoft.com/azure/azure-resource-manager/bicep/overview). The provided Bicep template is responsible for creating the Azure resources as well as executing the LogonScript (AKS Edge Essentials cluster creation, AIO deployment, Azure resource deployment and Azure Arc onboarding (Azure VM and AKS Edge Essentials cluster)) on the Azure VM.

Reviewer: ```suggestion
The following Jumpstart scenario will show how to create an AKS Edge Essentials cluster in Azure Windows Server VM and connect the Azure VM and AKS Edge Essentials cluster to Azure Arc using [Azure Bicep](https://learn.microsoft.com/azure/azure-resource-manager/bicep/overview). The provided Bicep template is responsible for creating the Azure resources as well as executing the LogonScript (AKS Edge Essentials cluster creation, AIO deployment, Azure resource deployment, and Azure Arc onboarding (Azure VM and AKS Edge Essentials cluster) on the Azure VM.
```

- Select My IP address from the dropdown.

  <img src="./16.png" alt="Screenshot showing adding a new allow RDP inbound security rule" width="400">

  <br/>

  ![Screenshot showing all inbound security rule](./18.png)

  ![Screenshot showing connecting to the VM using RDP](./19.png)

### Connect using Azure Bastion

- If you have chosen to deploy Azure Bastion in your deployment, use it to connect to the VM.

  ![Screenshot showing connecting to the VM using Bastion](./20.png)

  > **Note: When using Azure Bastion, the desktop background image is not visible. Therefore some screenshots in this guide may not exactly match your experience if you are connecting to *AIO-Client-VM* with Azure Bastion.**

### Connect using just-in-time access (JIT)

If you already have [Microsoft Defender for Cloud](https://docs.microsoft.com/azure/defender-for-cloud/just-in-time-access-usage?tabs=jit-config-asc%2Cjit-request-asc) enabled on your subscription and would like to use JIT to access the Client VM, use the following steps:

- In the Client VM configuration pane, enable just-in-time. This will enable the default settings.

  ![Screenshot showing how to enable JIT](./21.png)

>   ![Screenshot showing how to enable JIT](./22.png)

Reviewer: Missing step 2 in the img

  ![Screenshot showing AIO-Client-VM](./32.png)

  ![Screenshot showing AIO-Client-VM](./33.png)

  ![Screenshot showing AIO-Client-VM](./34.png)

  ![Screenshot showing AIO-Client-VM](./35.png)

  ![Screenshot showing AIO-Client-VM](./36.png)

  ![Screenshot showing AIO-Client-VM](./37.png)

  ![Screenshot showing AIO-Client-VM](./38.png)

  ![Screenshot showing AIO-Client-VM](./39.png)

> **Note: Occassionaly you might get the below error in deploying AIO, this is a known issue. The automation will retry the deployment and proceed.**

![Screenshot showing AIO-Client-VM](./40.png)

  ![Screenshot showing AIO-Client-VM](./41.png)

  ![Screenshot showing AIO-Client-VM](./42.png)

  ![Screenshot showing AIO-Client-VM](./43.png)

  ![Screenshot showing AIO-Client-VM](./44.png)

  ![Screenshot showing AIO-Client-VM](./45.png)

  ![Screenshot showing AIO-Client-VM](./46.png)

  ![Screenshot showing AIO-Client-VM](./47.png)

- Deployment is complete! Let's begin exploring how Contoso staff can start using the deployed solution.

  ![Screenshot showing complete deployment](./48.png)

  ![Screenshot showing AIO resources in Azure portal](./49.png)

## Industrial telemetry at the edge

> The automation deploys and configures a simulator that simulates data from various bakery equipment, such as ovens, dividers, and mixers, including a dedicated simulation for a fryer unit—a critical asset in the bakery's production line. The MQTT listener captures this data and funnels it into *InfluxDb* which is optimized for time-series data. Grafana, a leading open-source platform for monitoring and observability, taps into InfluxDb to render comprehensive dashboards and analytics, enabling the plant staff at Contoso to monitor and enhance the performance of the strawberry donut production line effectively. Those dashboards provide near real-time information and insights, often projected directly within the production line area, to enable immediate response and decision-making where it matters most.

Reviewer: Grafana product link

- Open the MQTT explorer desktop shortcut, it is already configured to connect to the MQTT listener on the cluster.

  ![Screenshot showing opening MQTT explorer on the desktop](./50.png)

- Once connected, you will start seeing simulated data being transmitted with various metrics from the plant assets.

  ![Screenshot showing the simulated data](./51.png)

  ![Screenshot showing the simulated data](./52.png)

  ![Screenshot showing the simulated data](./53.png)

- To enable the plant staff to monitor and visualize this data in real-time, open the Edge browser, expand InfluxDB in the Favorites Bar and select *AIO - InfluxDB*.

  ![Screenshot showing opening influx bookmark](./54.png)

- Login with the username admin and the Windows Administrator password provided when you created the deployment.

  ![Screenshot showing logging into InfluxDB](./55.png)

- Click on dashboards and select the "Contoso Bakery Strawberry Donut production line" dashboard.

>   ![Screenshot showing the grafana dashboard](./56.png)

Reviewer: Missing step 2 in 56.png

> - Contoso staff can start monitoring the strawberry donut production line. The most critical KPI featured is the Overall Equipment Effectiveness (OEE). The OEE indicator is color-coded for at-a-glance status updates: a green indicator signifies an OEE above the target range of 90%, symbolizing optimal performance. A yellow indicator represents an OEE between 80% to 90%, suggesting acceptable but suboptimal performance, and a red indicator flags any OEE below 80%, indicating a need for immediate attention.

Reviewer: I would consider changing this to a table format

   ![Screenshot showing the grafana dashboard](./57.png)

> - Additionally, the dashboard breaks down the components of OEE—Availability, Product Quality, and Performance—to provide a detailed analysis. A key focus is on Availability, with constant monitoring of Downtimes to identify and classify lost time reasons, such as equipment malfunctions or ingredient shortages.

Reviewer: ```suggestion
- Additionally, the dashboard breaks down the components of OEE; *Availability*, *Product Quality*, and *Performance*, to provide a detailed analysis. A key focus is on *Availability*, with constant monitoring of downtime to identify and classify lost time reasons, such as equipment malfunctions or ingredient shortages.
```

   ![Screenshot showing the grafana dashboard](./58.png)

- Users can also view real-time data on the current product being produced—strawberry donuts—as well as the active shift, which in this case is the morning shift.

   ![Screenshot showing the grafana dashboard](./59.png)

- Critical equipment, such as the fryer, is under special surveillance. The dashboard displays the current oil temperature against the target temperature, along with the oil level in the fryer, ensuring that any deviations from the norm can be quickly addressed.

> ## Advanced executive dashboards using Azure Data Explorer(ADX)

Reviewer: ```suggestion
## Advanced executive dashboards using Azure Data Explorer (ADX)
```

> The automation deploys an *Azure Data Explorer* cluster and provides a dashboard for regional executives and directors who require a broader view across multiple plants. The Azure IoT MQ in the local cluster transmits data to Azure Event Grid, which acts as a cloud-based MQTT broker. It receives operational data and routes it to Azure Event Grid topics, and from there to the Event Hub. Event Hub streamlines the data inflow, allowing Azure Data Explorer to ingest it. Using Kusto Query Language (KQL), Azure Data Explorer provides advanced analytics and custom dashboards that give executives insights into the performance of multiple plants which is crucial for strategic decision-making and long-term planning.

Reviewer: ```suggestion
The automation deploys an ADX cluster and provides a dashboard for Contoso's regional executives and directors who require a broader view across multiple plants. The Azure IoT MQ in the local cluster transmits data to Azure Event Grid, which acts as a cloud-based MQTT broker. It receives operational data and routes it to Azure Event Grid topics, and from there to the Event Hub. Event Hub streamlines the data inflow, allowing ADX to ingest it. Using Kusto Query Language (KQL), Azure Data Explorer provides advanced analytics and custom dashboards that give executives insights into the performance of multiple plants which is crucial for strategic decision-making and long-term planning.
```

A note here referring the user the the diagram at the beginning can help here.

### Manually import dashboards

> **Note:** If you used the [Azure Developer CLI (azd) method](../deployment/#deployment-via-azure-developer-cli) to deploy the scenario, you may skip this section as these reports are automatically imported for you during the automated deployment.

> Follow the below steps in order to view the dashboard reports, you will need to import it into ADX.
> 
> - On the Client VM, open Windows Explorer and navigate to folder *C:\AIO\adx_dashboard* folder. This folder contains an ADX dashboard report JSON file (*dashboard.json*) with the ADX URI updated when the deployment PowerShell logon script is completed.

Reviewer: ```suggestion
Follow the below steps in order to view the dashboard reports, you will need to import them into ADX.

- On the Client VM, open Windows Explorer and navigate to folder *C:\AIO\adx_dashboard* folder. This folder contains an ADX dashboard report JSON file (*dashboard.json*) with the ADX URI updated when the deployment PowerShell logon script is completed.
```

- Once you are logged in to ADX dashboards, click on Dashboards in the left navigation to import the dashboard report.

  ![Screenshot showing how to navigate to ADX dashboard reports](./61.png)

- Select "Import dashboard from file" to select previously copied file from the Client VM to your local machine or the *C:\AIO\adx_dashboard* folder on the Client VM.

  ![Screenshot showing the import dashboard file](./62.png)

- Choose to import the *dashboard.json* file.

  ![Screenshot showing the dashboard report JSON file to import](./63.png)

- Confirm the dashboard report name, accept the suggested name (or choose your own), and click Create.

  ![Screenshot showing the dashboard report name confirmation](./64.png)

- It might take a couple of minutes for the dashboard to display some data in the report after deployment. Click Save to save the dashboard report in ADX.

  ![Screenshot showing the empty data in orders dashboard report](./65.png)

> - Contoso staff can start levergaing the dashboards, for instance, a regional manager overseeing three plants might use the ADX dashboard to assess the collective OEE of these facilities. The dashboard aggregates essential metrics such as Overall OEE, Performance, and Availability, offering a consolidated view of productivity and efficiency. It highlights the most prevalent current shift's waste reason—in this case, 'Overcooked' and quantifies waste volumes alongside unplanned downtimes and their duration in minutes.
> 
> - Furthermore, these stakeholders often engage in comparative analysis. The ADX dashboard enables them to compare the performance across the different shifts operating within a single plant, discerning the OEE for primary products like donuts, bread, and pastries. It also facilitates a comparative review of the OEE across the three plants located in Seattle, Mexico City, and Miami. Quality indicators are also on display, ensuring products meet the required specifications in size and weight.
> 
>   ![Screenshot showing the OEE ADX dashboard](./66.png)
> 
> 
> - Critical quality-affecting factors, such as the oil temperature in fryers which can influence the color, size, or doneness of donuts, are also monitored. This level of detail provides executives with the data needed to make informed decisions, ensuring product consistency and operational excellence across the board.

Reviewer: ```suggestion
- Contoso staff can start leveraging the dashboards, for instance, a regional manager overseeing three plants might use the ADX dashboard to assess the collective OEE of these facilities. The dashboard aggregates essential metrics such as Overall OEE, Performance, and Availability, offering a consolidated view of productivity and efficiency. It highlights the most prevalent current shift's waste reason - in this case, 'Overcooked' and quantifies waste volumes alongside unplanned downtimes and their duration in minutes.

- Furthermore, these stakeholders often engage in comparative analysis. The ADX dashboard enables them to compare the performance across the different shifts operating within a single plant, discerning the OEE for primary products like donuts, bread, and pastries. It also facilitates a comparative review of the OEE across the three plants located in Seattle, Mexico City, and Miami. Quality indicators are also on display, ensuring products meet the required specifications in size and weight.

  ![Screenshot showing the OEE ADX dashboard](./66.png)

- Critical quality-affecting factors, such as the oil temperature in fryers which can influence the color, size, or doneness of donuts, are also monitored. This level of detail provides executives with the data needed to make informed decisions, ensuring product consistency and operational excellence across the board.
```

## Cleanup deployment

- To clean up your deployment, simply delete the resource group using Azure CLI or Azure portal.

  ```shell
  az group delete -n <name of your resource group>
  ```

  ![Screenshot showing az group delete](./67.png)

  ![Screenshot showing group delete from Azure portal](./68.png)

- If you used Azure Developer CLI to deploy then ```azd down``` can be used instead.

  ![Screenshot showing azd down](./69.png)

## Next steps

> If you still having issues with the deployment, please refer to the [Troubleshooting](../troubleshooting//) section. Otherwise, if you have additional questions or feedback, please refer to the [FAQ](../../../faq/) section.

Reviewer: See updated Agora docs for these relative paths. Also, where is the troubleshooting section?
